### PR TITLE
changed data grid sort page query logic, added sidebar drag drop, added viz config panel changes

### PR DESCRIPTION
**Pull request by paulstn**
### Description
changed data grid sort page logic, added sidebar drag drop, added viz config panel changes

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx [hunk 1]
```diff
@@ -117,8 +129,35 @@ export const Sidebar = (props: ISidebarProps) => {
...
+    if (
+      explorerFields.availableFields.length > 0 &&
+      explorerFields.selectedFields.length === 0 &&
+      allFields.length === 0
+    ) {
+      setAllFields(explorerFields.availableFields);
+    }
+  }, [explorerFields.availableFields]);
+
+  const onDragEnd = ({
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/sidebar/sidebar.tsx:132]: did we include drag and drop changes?

File: public/components/event_analytics/explorer/explorer.scss [hunk 2]
```diff
@@ -15,7 +15,7 @@
 }
 
 .mainContentTabs .euiResizableContainer {
-  height: calc(100vh - 298px);
+  height: calc(100vh - 194px);
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/explorer.scss:18]: did you test event page and possible resizing of the the page?

File: public/components/visualizations/visualization.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const Visualization = ({
     // Markdown, it does not depend on if there is data
     if (vis.id === VIS_CHART_TYPES.Text) return [true, ''];
 
-    if (isEmpty(series)) return [false, VISUALIZATION_ERROR.INVALID_DATA]; // series is required to any visualization type
```
Comment (mengweieric) [inline public/components/visualizations/visualization.tsx:32]: If the query is not valid and returns no data, what would it show?

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx [hunk 4]
```diff
@@ -535,49 +538,60 @@ export const DataConfigPanelItem = ({
...
-      </EuiFlexItem>
-    </>
+            <EuiSpacer size="s" />
+            <EuiTitle size="xxs">
+              <h3>Bucket Offset</h3>
+            </EuiTitle>
+            {getNumberField('bucketOffset')}
+          </>
+        )}
+        {/* <EuiSpacer size="m" /> */}
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: remove this if we don't need?

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.scss [hunk 5]
```diff
@@ -39,4 +39,50 @@
   &.showSecondary > .wizConfig__section {
     transform: translateX(-100%);
   }
+}
```
Comment (mengweieric) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.scss:42]: have we verified and changes on these will not affect core visualize?

File: public/components/event_analytics/explorer/visualizations/index.tsx [hunk 6]
```diff
@@ -93,16 +93,7 @@ export const ExplorerVisualizations = ({
...
-                {!isMarkDown && (
-                  <div
-                    className="explorer__vizDataConfig"
-                    data-test-subj="explorer__vizDataConfig-panel"
-                  >
-                    {renderDataConfigContainer()}
-                  </div>
-                )}
-              </div>
+              {!isMarkDown && <>{renderDataConfigContainer()}</>}
```
Comment (ps48) [inline public/components/event_analytics/explorer/visualizations/index.tsx:96]: nit. Can push empty fragment `<></>` inside renderDataConfigContainer() return statement. Will make this a bit a cleaner. 
```
{!isMarkDown && renderDataConfigContainer()}

File: public/components/event_analytics/explorer/explorer.scss (hunk 2 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/explorer.scss:18]: Yes, this value is derived from the size of all of the components on top (such as search bar, etc)

File: public/components/visualizations/visualization.tsx (hunk 3 above)
Comment (paulstn) [reply public/components/visualizations/visualization.tsx:32]: I assumed that an empty series would mean that data hasn't been put in yet. Shouldn't this be the case? For invalid queries there are other checks in this file that will display invalid data if invalid.

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx (hunk 4 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.tsx]: removed

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.scss (hunk 5 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_configurations_panel.scss:42]: To the best of my knowledge, it shouldn't. The only code in observability that has this classname was the code I've added

File: public/components/event_analytics/explorer/visualizations/index.tsx (hunk 6 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/visualizations/index.tsx:96]: ok changed

Comment (paulstn) [reply public/components/event_analytics/explorer/visualizations/index.tsx:96]: nvm keep getting a lot of errors even when i move it into the return statement of renderDataConfigContainer, will leave it as is

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx [hunk 7]
```diff
@@ -54,6 +54,18 @@ export const Sidebar = (props: ISidebarProps) => {
   const [showFields, setShowFields] = useState<boolean>(false);
   const [searchTerm, setSearchTerm] = useState<string>('');
 
+  // list of every single field object
+  const [allFields, setAllFields] = useState<IField[]>([]);
+
+  // method to return the type of a field from its name
+  const getFieldTypes = (newFieldName: string) => {
+    let fieldType: string = '';
+    allFields.map((field) => {
```
Comment (ps48) [inline public/components/event_analytics/explorer/sidebar/sidebar.tsx]: This is a quick fix. Ideally, we can do away with `allFields` and just loop over `explorerFields.availableFields` and `explorerFields.selectedFields` to get type.

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx (hunk 1 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/sidebar/sidebar.tsx:132]: It was mostly implemented already by Sumukh, all I did was add the logic for a field being added/removed. Your choice as to whether we implement or not, I can remove it

File: public/components/event_analytics/explorer/sidebar/sidebar.tsx (hunk 7 above)
Comment (paulstn) [reply public/components/event_analytics/explorer/sidebar/sidebar.tsx]: Ok I can make the change

File: public/components/visualizations/visualization.tsx (hunk 3 above)
Comment (mengweieric) [reply public/components/visualizations/visualization.tsx:32]: we can check with UX team on this

**Pull request merged**